x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -35,6 +41,61 @@ static DeprecationIssue checkProcessors(final Settings settings , final PluginsA
             "https://www.elastic.co/guide/en/elasticsearch/reference/7.4/breaking-changes-7.4.html#deprecate-processors");
     }
 
+    static DeprecationIssue checkMissingRealmOrders(final Settings settings, final PluginsAndModules pluginsAndModules) {
+        final String orderNotConfiguredRealms = RealmSettings.getRealmSettings(settings).entrySet()
+                .stream()
+                .filter(e -> Objects.isNull(e.getValue().get(RealmSettings.ORDER_SETTING_KEY)))
+                .map(e -> e.getKey().getName())
+                .collect(Collectors.joining("; "));
+
+        if (false == Strings.hasText(orderNotConfiguredRealms)) {
+            return null;
+        }
+
+        final String details = String.format(
+            Locale.ROOT,
+            "Found realms without order config: [%s]. In next major release, node will fail to start with missing realm order",
+            orderNotConfiguredRealms);
+        return new DeprecationIssue(
+            DeprecationIssue.Level.CRITICAL,
+            "Realm order will be required in next major release.",
+            "",
```

Comment:
Per comments on the PR itself, we'll need to provide a URL here.

Reply:
Updated the doc file and added the corresponding URL here.